core/headerchain.go

```diff
@@ -430,6 +430,26 @@ func (hc *HeaderChain) GetAncestor(hash common.Hash, number, ancestor uint64, ma
 	return hash, number
 }
 
+// GetHorizon returns the N-3 prime header number.
+func (hc *HeaderChain) GetHorizon() uint64 {
```

Comment:
IMO, this function does not actually get the horizon. We haven't agreed on a definition for what the horizon is, but I think we can all agree this isn't it, lol

For now just set the threshold to a large number -- 20? And create an issue to refactor this to implement a correct horizon check.